src/ir/ty.rs

```diff
@@ -1194,7 +1194,23 @@ impl Type {
         };
 
         let name = if name.is_empty() { None } else { Some(name) };
-        let is_const = ty.is_const();
```

Comment:
I think this works, but instead the better fix is (I think), changing the following line:

https://github.com/rust-lang-nursery/rust-bindgen/blob/515ca97b462f348b7b3bb9eaea98b2e0bb1b624b/src/codegen/mod.rs#L3057

To not check `self.is_const()`, pointing out that pointer-constness itself can't be represented in rust.

Or in any case that line could be cleaned up.

I think handling it in codegen is better though, since there are a couple other places where we use libclang's is_const in `context.rs`.

Also, could you test this with constants? This would solve #511. The test-case in that issue's description is pretty good.

Reply:
Mhhh. It's somewhat strange, I tried to apply your suggestion but I am getting test failures. It seems that `ctx.resolve_type(inner)` resolves to an opaque type? I am not sure though, I didn't have much time to debug this (I guess, I'll give it another shot later today).

But this PR should also fix #511 (Well, it seems like it).

Reply:
If you get stuck let me know, we can merge this for now and I can take a look at that more carefully. Please do file an issue for that if we leave it as a followup and add the tests for #511 if you can though.

Thank you!

Reply:
Yeah, I think that's best for now. I'll create a new issue after this PR is merged.
I've also pushed the test from issue #511.